### PR TITLE
docs: update docs about app router in use-with-next

### PR DESCRIPTION
**Pull request by kiner-tang**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [X] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution

- update docs about app router in use-with-next

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | update docs about app router in use-with-next |
| 🇨🇳 Chinese | 在 use-with-next 当中更新关于 app router 的文档  |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [X] Doc is updated/provided or not needed
- [X] Demo is updated/provided or not needed
- [X] TypeScript definition is updated/provided or not needed
- [X] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at dba9c4f</samp>

This pull request updates the documentation of using antd with Next.js App Router. It adds a new section to both the English and Chinese versions that shows how to use `@ant-design/cssinjs` to improve the first-screen rendering performance. It also provides example code and a GitHub link for reference.

### <samp>🤖 Generated by Copilot at dba9c4f</samp>

*  Add a new section to the documentation of using antd with Next.js App Router in both English and Chinese languages ([link](https://github.com/ant-design/ant-design/pull/43579/files?diff=unified&w=0#diff-f40a07db5e50bc714a69751497cc62a602271f477eabf7c895f2fa9950461ae4R51-R109), [link](https://github.com/ant-design/ant-design/pull/43579/files?diff=unified&w=0#diff-b532268d5c5f44bba9caadf328fe6567e31ba5fc7c36cfe6eec648bcf1b36489R51-R109))

### CODE REVIEW

File: docs/react/use-with-next.zh-CN.md [hunk 1]
```diff
@@ -48,3 +48,62 @@ export default Home;
...
+import React from 'react';
+import { useServerInsertedHTML } from 'next/navigation';
+import { StyleProvider, createCache, extractStyle } from '@ant-design/cssinjs';
+
+export default function StyledComponentsRegistry({ children }: { children: React.ReactNode }) {
+  const cache = createCache();
+
+  useServerInsertedHTML(() => {
+    return (
+      <>
```
Comment (li-jia-nan) [inline docs/react/use-with-next.zh-CN.md:74]: 这层标签是不是可以省略掉？

File: docs/react/use-with-next.zh-CN.md [hunk 2]
```diff
@@ -48,3 +48,62 @@ export default Home;
...
+
+import React from 'react';
+import { useServerInsertedHTML } from 'next/navigation';
+import { StyleProvider, createCache, extractStyle } from '@ant-design/cssinjs';
+
+export default function StyledComponentsRegistry({ children }: { children: React.ReactNode }) {
+  const cache = createCache();
+
+  useServerInsertedHTML(() => {
+    return (
```
Comment (li-jia-nan) [inline docs/react/use-with-next.zh-CN.md:73]: 另外是箭头函数，这个 return 是不是也可以省略掉？

Comment (kiner-tang) [reply docs/react/use-with-next.zh-CN.md:73]: 可以

Comment (kiner-tang) [reply docs/react/use-with-next.zh-CN.md]: done. 另外示例那边也改了一下：https://github.com/ant-design/ant-design-examples/pull/21

**Comment by MadCcc** [comment]
pages 的写法已在这里写一下

**Comment by kiner-tang** [comment]
> pages 的写法也在这里写一下

好的

File: docs/react/use-with-next.zh-CN.md [hunk 3]
```diff
@@ -48,3 +48,82 @@ export default Home;
...
+```
+
+4. 在页面中使用
+
+```tsx
+'use client';
+
+import React from 'react';
+import { Button } from 'antd';
+import withTheme from '../../theme';
```
Comment (li-jia-nan) [inline docs/react/use-with-next.zh-CN.md:112]: 这个 `withTheme` 是怎么实现的？

Comment (kiner-tang) [reply docs/react/use-with-next.zh-CN.md:112]: 这个我也在文档说明一下吧

**Review by zombieJ — APPROVED** [review]

File: docs/react/use-with-next.zh-CN.md [hunk 4]
```diff
@@ -48,3 +48,115 @@ export default Home;
...
+    }}
+  >
+    {/* nesting */}
+    <ConfigProvider
+      theme={{
+        token: {
+          borderRadius: 16,
+        },
+      }}
+    >
```
Comment (afc163) [inline docs/react/use-with-next.zh-CN.md:129]: 没必要嵌套两层吧，应该推荐用户定义一个就行。

File: docs/react/use-with-next.zh-CN.md [hunk 5]
```diff
@@ -48,3 +48,115 @@ export default Home;
...
+```
+
+5. 在页面中使用
+
+```tsx
+'use client';
+
+import React from 'react';
+import { Button } from 'antd';
+import withTheme from '../../theme';
```
Comment (afc163) [inline docs/react/use-with-next.zh-CN.md:145]: 可以把 theme 对象单独定义一个文件 `theme.ts` 引入就好，这样主题文件只包含定制信息。

File: docs/react/use-with-next.zh-CN.md (hunk 4 above)
Comment (li-jia-nan) [reply docs/react/use-with-next.zh-CN.md:129]: - #43590

File: docs/react/use-with-next.zh-CN.md (hunk 5 above)
Comment (li-jia-nan) [reply docs/react/use-with-next.zh-CN.md:145]: - #43590

**Pull request merged**